share/functions/pushd.fish

```diff
@@ -76,6 +76,6 @@ function pushd --description 'Push directory to stack'
     end
 
     # argv[1] is a directory
-    set -g -p dirstack $PWD
     cd $argv[1]
+    and set -g -p dirstack $PWD
```

Comment:
Ummmm... sorry. This wants the directory *before* changing.

So you'll have to store $PWD, and only add it to dirstack if the cd succeeds.

Reply:
whoops, fixed